api/routes/index.js

```diff
@@ -2,16 +2,165 @@ const express = require('express');
 const db = require('../../db/index');
 const router = express.Router();
 
-// This is an example SQL query - use this as a template
-router.get('/', async (req, res) => {
+/** Returns an array of tags given a SQL query
+ * map through an array,
+ * grab just the values of each dictionary,
+ *  and flatten the resulting array
+ */
+function getArrayOfTags(query) {
+  return query.rows.map(Object.values).flat();
+}
+
+/** Returns the difference between two arrays
+ *  Returns an array of objects in Array1
+ *  That are not in Array2
+ */
+
+function diffArray(arr1, arr2) {
+  var n = arr1.filter(x => !arr2.includes(x));
+  return n;
+}
+
+router.get('/coop/:CoopID', async (req, res) => {
   try {
-    const query = await db.query(
-      `SELECT 'This is the root of indexRouter.' as text;`
-    );
-    res.send(query.rows[0]['text']);
+    const id = req.params['CoopID'];
+    var command =
+      'SELECT tag_name FROM coop_tags JOIN tags ON coop_tags.tag_id = tags.id WHERE coop_tags.coop_id= $1';
+    var values = [id];
+
+    var queryTags = await db.query(command, values);
+    var listOfTags = getArrayOfTags(queryTags);
+
+    var command = 'SELECT * FROM coops WHERE id = $1';
+    const query = await db.query(command, values);
+
+    if (query.rows.length >= 1) {
+      query.rows[0]['tags'] = listOfTags;
+    } else {
+      res.status(404).send({ error: `coop ${id} not found` });
+    }
+
+    res.send(query.rows[0]);
   } catch (error) {
     console.log(error.stack);
   }
 });
 
+router.post('/coop', async (req, res) => {
+  var idQuery = await db.query('SELECT id FROM coops ORDER BY id DESC LIMIT 1');
+  if (getArrayOfTags(idQuery).length >= 1) {
+    var id = getArrayOfTags(idQuery)[0] + 1;
+  } else {
+    var id = 1;
+  }
+
+  const { email, name, addr, pass } = req.body;
+
+  const text =
+    'INSERT INTO coops(id, email, pass, coop_name, addr) VALUES($1, $2, $3, $4, $5)';
+  const values = [id, email, pass, name, addr];
```

Comment:
why is the `id` still manually being calculated? `id` is a [SERIAL](https://www.postgresqltutorial.com/postgresql-serial/) type so you don't need to explicitly set it when you insert the row
```javascript
await db.query(
  'INSERT INTO coops (email, pass, coop_name, addr) VALUES($1, $2, $3, $4);',
  [email, pass, name, addr]
);

Reply:
see the comment below on `cofed.sql` to see what's likely causing the problem